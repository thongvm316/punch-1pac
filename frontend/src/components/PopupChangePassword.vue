<template>
  <div class="modal" :class="{ active: open }">
    <span class="modal-overlay" @click="remind"></span>
    <div class="modal-container">
      <div class="modal-header">
        <button ref="btnCloseModal" class="btn btn-clear float-right" @click="remind"></button>
        <h3 class="modal-title">{{ $t('popup.changePassword.title') }}</h3>
      </div>
      <div class="modal-body">
        <div class="content">{{ $t('popup.changePassword.description') }}</div>
      </div>
      <div class="modal-footer">
        <router-link ref="btnChangePassword" class="btn btn-success" to="/settings/password" @click.native="remind">{{ $t('popup.changePassword.btnChange') }}</router-link>
        <button ref="btnCancel" type="button" class="btn" @click="remind">{{ $t('popup.changePassword.btnRemind') }}</button>
      </div>
    </div>
  </div>
</template>

<script>
import { mapMutations } from 'vuex'
import { INITIAL_STATES_UPDATE_PASSWORD_CHANGED } from '../store/mutation-types.js'

export default {
  name: 'popup-change-password',

  data() {
    return {
      open: false
    }
  },

  methods: {
    ...mapMutations('initialStates', [INITIAL_STATES_UPDATE_PASSWORD_CHANGED]),

    remind() {
      this[INITIAL_STATES_UPDATE_PASSWORD_CHANGED](true)
      this.open = !this.currentUser.password_changed
    }
  },

  created() {
    this.open = !this.currentUser.password_changed
  }
}
</script>
