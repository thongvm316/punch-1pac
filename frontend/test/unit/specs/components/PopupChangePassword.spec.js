import { shallowMount } from '@vue/test-utils'
import wrapperOps from '../../supports/wrapper'
import PopupChangePassword from '@/components/PopupChangePassword'

const remind = jest.spyOn(PopupChangePassword.methods, 'remind')
const INITIAL_STATES_UPDATE_PASSWORD_CHANGED = jest.spyOn(PopupChangePassword.methods, 'INITIAL_STATES_UPDATE_PASSWORD_CHANGED')

const localWrapperOps = {
  ...wrapperOps,
  computed: {
    currentUser () {
      return {
        password_changed: false
      }
    }
  }
}

describe('PopupChangePassword.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PopupChangePassword, localWrapperOps)
  })

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when PopupChangePassword was mounted ', () => {
    it('should render correctly', () => {
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.isVueInstance()).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when methods', () => {
    it('should remind toHaveBeenCalled', () => {
      wrapper.find({ ref: 'btnCloseModal' }).trigger('click')

      expect(remind).toHaveBeenCalled()
      expect(INITIAL_STATES_UPDATE_PASSWORD_CHANGED).toHaveBeenCalledWith(true)
    })
  })
})
