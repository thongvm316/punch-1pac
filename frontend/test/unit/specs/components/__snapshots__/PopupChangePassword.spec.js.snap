// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupChangePassword.vue when PopupChangePassword was mounted  should render correctly 1`] = `
<div class="modal active"><span class="modal-overlay"></span>
  <div class="modal-container">
    <div class="modal-header"><button class="btn btn-clear float-right"></button>
      <h3 class="modal-title">
        Your password is not secured
      </h3>
    </div>
    <div class="modal-body">
      <div class="content">
        You are using password generated by system, it's not secured. Please change your password as soon as possible
      </div>
    </div>
    <div class="modal-footer">
      <router-link-stub to="/settings/password" tag="a" event="click" class="btn btn-success">
        Change password
      </router-link-stub> <button type="button" class="btn">
        Remind me later
      </button>
    </div>
  </div>
</div>
`;
